src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -20,6 +20,16 @@
     <BuildArgs>$(BuildArgs) /p:SourceBuiltShippingPackagesDir=$(RepoArtifactsShippingPackagesDir)</BuildArgs>
     <!-- Trim the trailing slash as it breaks argument parsing on Windows if this is the last argument. -->
     <BuildArgs>$(BuildArgs) /p:SourceBuiltNonShippingPackagesDir=$(RepoArtifactsNonShippingPackagesDir.TrimEnd('\'))</BuildArgs>
+
+    <!-- Pass RID and Portable/non-portable information -->
+    <_platformIndex>$(NETCoreSdkRuntimeIdentifier.LastIndexOf('-'))</_platformIndex>
+    <RuntimeOS>$(NETCoreSdkRuntimeIdentifier.Substring(0, $(_platformIndex)))</RuntimeOS>
+
+    <_platformIndex>$(NETCoreSdkPortableRuntimeIdentifier.LastIndexOf('-'))</_platformIndex>
+    <BaseOS>$(NETCoreSdkPortableRuntimeIdentifier.Substring(0, $(_platformIndex)))-$(TargetArchitecture)</BaseOS>
+    <BuildArgs>$(BuildArgs) /p:PortableBuild=$(PortableBuild)</BuildArgs>
+    <BuildArgs Condition="'$(ShortStack)' != 'true' and '$(TargetOS)' != 'linux-musl'">$(BuildArgs) /p:RuntimeOS=$(RuntimeOS)</BuildArgs>
+    <BuildArgs Condition="'$(ShortStack)' != 'true' and '$(TargetOS)' != 'linux-musl'">$(BuildArgs) /p:BaseOS=$(BaseOS)</BuildArgs>
```

Comment:
These properties don't make sense in most of the VMR repositories (i.e. managed only repos). Do we really want to always pass them in?

Reply:
It might make sense to introduce a new property that defines if these properties should be passed in. That would then be set in runtime.proj and aspnetcore.proj. Btw I would expect a few others to need this as well eventually.

Reply:
My thinking here was to make these properties available in the same way that the Arcade SDK makes the other properties used by the Installers package are always available even if nothing uses them.

Putting the properties here also helps reduce the number of multi-repo changes we'd need to make when we inevitably forget that we need to opt-in here.

I can make it opt-in though if you prefer (and opt-in runtime, aspnetcore, SDK)

Reply:
Global properties are different though... They appear in the repo build invocation and in the repo binlog as build args. Keeping the number of global properties limited is a general recommendation.

Here with an orchestrator, I understand that it doesn't matter that much but still wanted to mention it.

My next question would probably be why we even need to pass some of these in instead of calculating them in the repo itself. We eventually want to set up repo UB legs similar to the existing repo source build legs.

Reply:
These properties are calculating the default values for `HostOS` and `BaseOS`. In source-build scenarios (the only scenario where they really matter today), they'll be overridden on the command line by the orchestrator.

What do you think about moving these "default" definitions into the Arcade SDK and then only pass them here when they're explicitly provided?

Reply:
Yes that would be best. But we can for sure so that in a follow-up. Feel free to merge this one when ready.